File: ios/GoogleAdsWebviewViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(GoogleAdsWebviewViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(adHost, NSString)
RCT_EXPORT_VIEW_PROPERTY(adSlot, NSString)
RCT_EXPORT_VIEW_PROPERTY(adClient, NSString)
RCT_EXPORT_VIEW_PROPERTY(pageUrl, NSString)
RCT_EXPORT_VIEW_PROPERTY(onUnfilledAd, RCTBubblingEventBlock)

@end

File: ios/GoogleAdsWebviewViewManager.swift
@objc(GoogleAdsWebviewViewManager)
class GoogleAdsWebviewViewManager: RCTViewManager {
    
    override func view() -> GoogleAdsWebviewView {
        return GoogleAdsWebviewView()
    }
    
    @objc override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
